src/Tokenizer/AbstractTypeTransformer.php

```diff
@@ -21,76 +21,47 @@
  */
 abstract class AbstractTypeTransformer extends AbstractTransformer
 {
+    private const TYPE_END_TOKENS = [')', [T_CALLABLE], [T_NS_SEPARATOR], [T_STRING], [CT::T_ARRAY_TYPEHINT]];
```

Comment:
I hope our tests are well enough to ensure we don't break anything accidentally with change in this file